### PR TITLE
REST API Pagination

### PR DESCRIPTION
**Pull request by MtGuerenS**
Issue #125

Establishes limit/offset pagination. Limit can be specified in .env and offset can be specified in query. Default value for limit is 20 and offset is 0. Limit/offset was applied to:

- Section/Professor/Degree/Exam/Section...Search()
- ExamAll()

### CODE REVIEW

File: api/configs/env.go [hunk 1]
```diff
@@ -15,3 +16,18 @@ func EnvMongoURI() string {
...
 	return os.Getenv("MONGODB_URI")
 }
+
+func EnvLimit() int64 {
+	err := godotenv.Load()
+	if err != nil {
+		log.Fatal("Error loading .env file: %v", err)
+	}
+
+	limit, err := strconv.ParseInt(os.Getenv("LIMIT"), 10, 64)
```
Comment (ahshabbir) [inline api/configs/env.go:26]: Should this be an argument or an environment variable?

File: api/controllers/course.go [hunk 2]
```diff
@@ -35,8 +37,22 @@ func CourseSearch() gin.HandlerFunc {
 			query[key] = c.Query(key)
 		}
 
+		delete(query, "offset") // offset not in query because it is for pagination not searching
```
Comment (ahshabbir) [inline api/controllers/course.go:40]: Is "offset" guaranteed to be in query?

**Review by ahshabbir — CHANGES_REQUESTED** [review]
General Comments:
- Instead of doing duplicate work across tge controllers, is it possible to setup the options once and pass them in as an argument? This might require some design reassesment @AdamMcAdamson 
- This would probably warrent a test to ensure pagination works as expected

File: api/controllers/course.go (hunk 2 above)
Comment (MtGuerenS) [reply api/controllers/course.go:40]: No. If "offset" is not in query, then no change occurs when performing delete(query, "offset).

File: api/configs/env.go (hunk 1 above)
Comment (MtGuerenS) [reply api/configs/env.go:26]: Environment variable I think, what would be the benefit of making it an argument?

**Comment by MtGuerenS** [comment]
For "setup the options once and pass them in as an argument", could this be what you mean?
Screenshot 2022-10-27 214919
^^ In package configs ^^
Screenshot 2022-10-27 214854
Tested it, and it seems to work like the original code.

**Review by ahshabbir — CHANGES_REQUESTED** [review]
Needs formatting adjustments and some sort of automated test (integration test seems appropriate).

**Review by AdamMcAdamson — APPROVED** [review]
Great work!

**Pull request merged**